src/snowflake/cli/plugins/nativeapp/utils.py

```diff
@@ -81,3 +81,33 @@ def verify_exists(paths_to_sync: List[Path]):
     for path in paths_to_sync:
         if not path.exists():
             raise ClickException(f"The following path does not exist: {path}")
+
+
+def get_all_file_paths_under_dir(directory: Path):
+    """
+    Recursively traverse the given directory and yield file paths. If the directory is a symlink, then the function follows the behavior of os.walk for symlinks.
+    The directory may be a relative path to the project root.
+    """
+    for root, _, files in os.walk(directory):
+        for file in files:
+            yield Path(os.path.join(root, file))
+
+
+def is_python_file(file_path: Path):
+    """
+    Checks if the given file is a python file.
+    """
+    return file_path.suffix == ".py"
+
+
+def filter_files(generator: Generator, predicate_func: Callable):
+    """
+    Filter file paths based on a given predicate function.
+    """
+    for file_path in generator:
+        if predicate_func(file_path):
+            yield file_path
+
+
+def is_single_quoted(name: str) -> bool:
```

Comment:
the purpose of single quoting is ambiguous then. Snowflake has a specific syntax for single quoted strings, including how to escape nesting quotation marks. I'm not super comfortable with this helper, it's too generic/ambiguous and could be misused. I'd move it to your own module, at the very least, instead of making it a general utility.